vllm/attention/backends/flashinfer.py

```diff
@@ -247,19 +258,23 @@ def get_graph_input_buffers(self,
             "slot_mapping": attn_metadata.slot_mapping,
         }
 
-    def prepare_graph_input_buffers(self,
-                                    input_buffers,
-                                    attn_metadata,
-                                    is_encoder_decoder_model: bool = False):
+    def prepare_graph_input_buffers(
+        self,
+        input_buffers,
+        attn_metadata,
+        is_encoder_decoder_model: bool = False,
+    ):
         return
 
     def begin_forward(self, model_input):
         assert not self._is_graph_capturing
         state = self
-        if model_input.attn_metadata.use_cuda_graph:
+        use_cuda_graph = model_input.attn_metadata.use_cuda_graph
+        is_decode = model_input.attn_metadata.num_prefills == 0
+        if use_cuda_graph and is_decode:
```

Comment:
Can you add comment about why we look at use_cuda_graph and is_decode to update the state?